commands/utility/removeinfraction.js

```diff
@@ -0,0 +1,114 @@
+const Discord = require('discord.js');
+const dateFormat = require('dateformat');
+
+module.exports = {
+  name: 'removeinfraction',
+  description: 'Removes a specific infraction based on the ID provided',
+
+  execute(msg, con, args) {
+    const {status, err, userInfraction, infractionID} = validInfraction(
+      msg,
+      args
+    );
+    if (!status) {
+      return msg.reply(err);
+    }
+
+    infractionSQL(msg, userInfraction, infractionID, args, con);
+  },
+};
+
+function validInfraction(msg, args) {
+  const data = {
+    status: false,
+    err: null,
+    userInfraction: null,
+    infractionID: null,
+  };
+
+  if (
+    !msg.member.roles.cache.some(
+      (role) => role.name === 'Admin' || role.name === 'Moderator'
+    )
+  ) {
+    data.err = msg.reply('You must be an Admin to use this command.');
+    return data;
+  }
+
+  data.userInfraction =
+    msg.mentions.members.first() || msg.guild.members.cache.get(args[0]);
+  if (!data.userInfraction) {
+    data.err = 'Please provide a user to remove this infraction from.';
+    return data;
+  }
+  console.log(args);
+  data.infractionID = args[1];
+  if (data.infractionID === '') {
+    data.err = 'Please provide a infraction ID to remove.';
+    return data;
+  }
+
+  data.status = true;
+  return data;
+}
+
+function infractionSQL(msg, userInfraction, infractionID, args, con) {
+  const now = new Date();
+  const date = dateFormat(now, 'yyyy-mm-dd HH:MM:ss');
+
+  const action = 'cc!removeinfraction ' + args.join(' ');
+
+  // Inserts row into database
+  const sql = `UPDATE infractions SET valid = ? WHERE id = ?;
+    INSERT INTO mod_log (timestamp, moderator, action, length_of_time, reason) VALUES
+    (?, ?, ?, NULL, NULL);`;
+
+  const values = [
+    false,
+    infractionID,
+    date,
+    msg.author.id,
+    action,
+    infractionID,
+  ];
+  const escaped = con.format(sql, values);
+
+  con.query(escaped, function (err, result) {
+    if (err) {
+      console.log(err);
+    } else {
+      console.log('Infraction was set to invalid');
```

Comment:
I think this should only be logged if the infraction was actually set to invalid (right now, it logs even if the infraction ID is invalid).